### PR TITLE
feat: implement basic flow reporting via server-sent events

### PR DESCRIPTION
**Pull request by BastiDood**
[Using server-sent events (SSE)](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events), we are able to send to the client live updates about flow reports, leaks, shutdowns, and resets. This PR only implements flow reports so far, but the necessary infrastructure is finally in place that enables us to send arbitrary JSON (unidirectionally) to one or more clients subscribed to the same MAC address.

Here, we revamp the `GET /api/metrics` endpoint to now require one new query parameter: `start`, which is a Unix timestamp (based on UTC) that lets the client specify the "start date" of the first timestamps. Once the SSE stream has been established, the client will receive the first batch of timestamps _en masse_ before receiving live updates one by one.

### CODE REVIEW

File: src/database.rs [hunk 1]
```diff
@@ -1,12 +1,13 @@
-pub use model::{report::Flow, MacAddress};
-pub use tokio_postgres::Client;
-pub use uuid::Uuid;
-
 use crate::model::ClientFlow;
 
+use chrono::{DateTime, Utc};
```
Comment (jkrperson) [inline src/database.rs:3]: This means we're removing `chrono::` on the other functions.

Comment (BastiDood) [reply src/database.rs:3]: If I'm not mistaken, I believe Rust Analyzer automatically handled this for me.

Comment (BastiDood) [reply src/database.rs:3]: Oh, never mind... Lemme go fix that.

File: src/lib.rs [hunk 2]
```diff
@@ -4,4 +4,4 @@ extern crate alloc;
 
 pub mod database;
 pub mod router;
-pub mod model;
\ No newline at end of file
+pub mod model;
```
Comment (jkrperson) [inline src/lib.rs:7]: Nothing changed here.

Comment (BastiDood) [reply src/lib.rs:7]: Actually, `cargo fmt` added a missing newline there so that it's LF-compliant.

Comment (jkrperson) [reply src/lib.rs:7]: Got it! Approved!

**Review by jkrperson — APPROVED** [review]
Looks good!

**Pull request merged**